.gitlab/binary_build/system_probe.yml

```diff
@@ -47,6 +42,13 @@ build_system-probe-x64:
   extends: .system-probe_build_common
   variables:
     ARCH: amd64
+  before_script:
+    # Retrieve nikos from S3
+    - $S3_CP_CMD $S3_PERMANENT_ARTIFACTS_URI/nikos-$ARCH.tar.gz /tmp/nikos.tar.gz
+    - mkdir -p $NIKOS_EMBEDDED_PATH
+    - tar -xf /tmp/nikos.tar.gz -C $NIKOS_EMBEDDED_PATH
```

Comment:
This cannot work because it overrides the `before_script` defined in `.system-probe_build_common` (if two `build_script` sections are defined for a job, they are not merged). I believe you need to put these three lines in the `script` section below.